libnmstate/nm/nmclient.py

```diff
@@ -116,14 +117,29 @@ def run(self, timeout):
 
     @property
     def cancellable(self):
-        return self._cancellable
+        return self._cancellables[0]
+
+    def new_cancellable(self):
+        c = Gio.Cancellable.new()
+        self._cancellables.append(c)
+        return c
+
+    def drop_cancellable(self, c):
+        idx = self._cancellables.index(c)
+        if idx == 0:
+            raise NmstateBug("Cannot drop main cancellable")
```

Comment:
Single quotes please.
Regarding the exception name, how about: `MainloopCancellableDropError` ?